test/quic-openssl-docker/run_endpoint.sh

```diff
@@ -35,25 +35,51 @@ if [ "$ROLE" == "client" ]; then
     rm -f $CURLRC 
 
     case "$TESTCASE" in
-    "http3"|"transfer")
+    "http3")
         echo -e "--verbose\n--parallel" >> $CURLRC
         generate_outputs_http3
         dump_curlrc
-        SSL_CERT_FILE=/certs/ca.pem curl --config $CURLRC || exit 1
+        SSL_CERT_FILE=/certs/ca.pem curl --config $CURLRC 
+        if [ $? -ne 0 ]
+        then
+            exit 1
+        fi
         exit 0
         ;;
-    "handshake")
-       OUTFILE=$(basename $REQUESTS)
-       echo -e "--verbose\n--http3\n-H \"Connection: close\"\n-o /downloads/$OUTFILE\n--url $REQUESTS" >> $CURLRC
-       dump_curlrc
-       SSL_CERT_FILE=/certs/ca.pem curl --config $CURLRC || exit 1
+    "handshake"|"transfer"|"retry")
+       HOSTNAME=none
+       for req in $REQUESTS
+       do
+           OUTFILE=$(basename $req)
```

Comment:
This assumes that the `path` of the request URLs is always a single element deep (`/foo[?params]` not `/foo/bar[?params]`. Is that a guaranteed feature of the tests?  Otherwise, `basename` throws away too much.

Reply:
In the test harness, yes, it is currently guaranteed.  Certainly not guaranteed for general use, but I think thats ok for a demo application

Reply:
No worries, but if you wanted to equivocally "unask" the question, it would be:

```
           OUTFILE=$(printf "%s\n" "$req" | sed -ne 's,^[^:]*://[^/]*,,p')
```